### PR TITLE
proto/lint: Add protobuf GitHub action and linter

### PR DESCRIPTION
**Pull request by thrasher-**
# PR Description

Adds a new GitHub action for protobuf linting and generation based on a build issue a user specified on Slack. This PR introduces the following;

1) `buf generate` to automate builds which detected duplicate gRPC reserve proxy endpoints, this breaks builds on Linux currently when using Buf.
2) Standardisation of proto spec variable names (snake case)
3) Proto file formatting similar to `go fmt`

I have temporarily turned off some of the linters until versioning is implemented.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## How has this been tested

- [x] go test ./... -race
- [x] golangci-lint run
- [x] proto-checks passing

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/proto-lint.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+          go-version: 1.18.x
+
+      - name: Setup build depends
+        run: |
+         go get github.com/grpc-ecosystem/grpc-gateway/v2/internal/descriptor@v2.10.0
+         go get google.golang.org/grpc/cmd/protoc-gen-go-grpc
+         go install github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-grpc-gateway
+         go install github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2
+         go install google.golang.org/protobuf/cmd/protoc-gen-go
+         go install google.golang.org/grpc/cmd/protoc-gen-go-grpc
```
Comment (gloriousCode) [inline .github/workflows/proto-lint.yml:24]: Should these be part of the readme?

**Review by gloriousCode — COMMENTED** [review]
Thanks for this! I appreciate linters for it, and the standardisation of our `proto` vars. I can build and it all still works. Just one question

File: engine/rpcserver.go [hunk 2]
```diff
@@ -2544,7 +2544,7 @@ func (s *RPCServer) GCTScriptStatus(_ context.Context, _ *gctrpc.GCTScriptStatus
 			return false
 		}
 		resp.Scripts = append(resp.Scripts, &gctrpc.GCTScript{
-			UUID:    vm.ID.String(),
+			Uuid:    vm.ID.String(),
```
Comment (shazbert) [inline engine/rpcserver.go:2547]: 😭

File: gctrpc/rpc.pb.go [hunk 3]
```diff
@@ -1,7 +1,7 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
-// 	protoc-gen-go v1.26.0
-// 	protoc        v3.11.2
+// 	protoc-gen-go v1.28.0
+// 	protoc        (unknown)
```
Comment (shazbert) [inline gctrpc/rpc.pb.go:4]: Things your learn [wow](https://github.com/bufbuild/buf/issues/799)

File: .github/workflows/proto-lint.yml (hunk 1 above)
Comment (thrasher-) [reply .github/workflows/proto-lint.yml:24]: The core ones required are already mentioned in: https://github.com/thrasher-corp/gocryptotrader/tree/master/gctrpc#installation and the same as the official docs: https://github.com/grpc-ecosystem/grpc-gateway#installation

Issuing these commands will report any dependant ones like the descriptor which can be left to the user to resolve and can be updated periodically

File: gctrpc/rpc.proto [hunk 4]
```diff
@@ -1,1720 +1,1718 @@
...
-    message Orders {
-        string exchange = 1;
-        map<string, string> order_status = 2;
-    }
-    repeated Orders orders = 1;
-    int64 count = 2;
+  message Orders {
+    string exchange = 1;
+    map<string, string> order_status = 2;
+  }
```
Comment (shazbert) [inline gctrpc/rpc.proto:422]: We could probably share this type with CancelBatchOrdersResponse?

**Review by shazbert — COMMENTED** [review]
Pretty much a tACK, everything works fine, one suggestion if you want to do it. I tested all the endpoints again that were now included in swagger just to make sure.

**Review by shazbert — APPROVED** [review]
tACK! clickity clack.

**Review by gloriousCode — APPROVED** [review]
tACK! Thanks for clarifying things

**Pull request merged**